projects/plugins/jetpack/_inc/client/recommendations/constants.js

```diff
@@ -14,8 +24,129 @@ export const RECOMMENDATION_WIZARD_STEP = {
+
+export const ONBOARDING_JETPACK_COMPLETE = 'JETPACK_COMPLETE';
+export const ONBOARDING_JETPACK_SECURITY = 'JETPACK_SECURITY';
+export const ONBOARDING_JETPACK_BACKUP = 'JETPACK_BACKUP';
+export const ONBOARDING_JETPACK_ANTI_SPAM = 'JETPACK_ANTI_SPAM';
+export const ONBOARDING_JETPACK_VIDEOPRESS = 'JETPACK_VIDEOPRESS';
+export const ONBOARDING_JETPACK_SEARCH = 'JETPACK_SEARCH';
+export const ONBOARDING_JETPACK_SCAN = 'JETPACK_SCAN';
+
+export const ONBOARDING_SUPPORT_START_TIMESTAMP = 1664323200000; // 2022-09-28
+
+export const ONBOARDING_ORDER = [
+	ONBOARDING_JETPACK_COMPLETE,
+	ONBOARDING_JETPACK_SECURITY,
+	ONBOARDING_JETPACK_BACKUP,
+	ONBOARDING_JETPACK_ANTI_SPAM,
+	ONBOARDING_JETPACK_VIDEOPRESS,
+	ONBOARDING_JETPACK_SEARCH,
+	ONBOARDING_JETPACK_SCAN,
+];
+
+export const ONBOARDING_NAME_BY_PRODUCT_SLUG = {
+	[ ONBOARDING_JETPACK_COMPLETE ]: JETPACK_COMPLETE_BUNDLES,
+	[ ONBOARDING_JETPACK_SECURITY ]: JETPACK_SECURITY_BUNDLES,
+	[ ONBOARDING_JETPACK_BACKUP ]: JETPACK_BACKUP_PRODUCTS,
+	[ ONBOARDING_JETPACK_ANTI_SPAM ]: JETPACK_ANTI_SPAM_PRODUCTS,
+	[ ONBOARDING_JETPACK_VIDEOPRESS ]: JETPACK_VIDEOPRESS_PRODUCTS,
+	[ ONBOARDING_JETPACK_SEARCH ]: JETPACK_SEARCH_PRODUCTS,
+	[ ONBOARDING_JETPACK_SCAN ]: JETPACK_SCAN_PRODUCTS,
+};
+
+export const SUMMARY_SECTION_BY_ONBOARDING_NAME = {
+	[ ONBOARDING_JETPACK_COMPLETE ]: {
+		name: 'Complete',
+		slugs: [
+			'backup-activated',
+			'scan-activated',
+			'antispam-activated',
+			'videopress-activated',
+			'search-activated',
+		],
+	},
+	[ ONBOARDING_JETPACK_SECURITY ]: {
+		name: 'Security',
+		slugs: [ 'backup-activated', 'scan-activated', 'antispam-activated' ],
+	},
+	[ ONBOARDING_JETPACK_BACKUP ]: {
+		name: 'Backup',
+		slugs: [ 'backup-activated' ],
+	},
+	[ ONBOARDING_JETPACK_ANTI_SPAM ]: {
+		name: 'Anti-Spam',
+		slugs: [ 'antispam-activated' ],
+	},
+	[ ONBOARDING_JETPACK_VIDEOPRESS ]: {
+		name: 'VideoPress',
+		slugs: [ 'videopress-activated' ],
+	},
+	[ ONBOARDING_JETPACK_SEARCH ]: {
+		name: 'Search',
+		slugs: [ 'search-activated' ],
+	},
+	[ ONBOARDING_JETPACK_SCAN ]: {
+		name: 'Scan',
+		slugs: [ 'scan-activated' ],
+	},
+};
+
+/**
+ * Function to get an onboarding for the specific product
+ *
+ * @param {string} productSlug - slug of the product
+ * @returns {string} onboarding name or null if onboarding not found
+ */
+export function getOnboardingNameByProductSlug( productSlug ) {
+	const foundIndex = Object.values( ONBOARDING_NAME_BY_PRODUCT_SLUG ).findIndex( slugs =>
+		slugs.includes( productSlug )
+	);
+
+	if ( -1 === foundIndex ) {
+		return null;
+	}
+
+	return Object.keys( ONBOARDING_NAME_BY_PRODUCT_SLUG )[ foundIndex ];
+}
+
+/**
+ * Function to get an onboarding priority
+ *
+ * @param {string} name - onboarding name
+ * @returns {number} the onboarding priority or 999 (max priority) if onboarding not found
+ */
+export function getOnboardingPriority( name ) {
+	const index = ONBOARDING_ORDER.indexOf( name );
+
+	return index > -1 ? index : 999;
```

Comment:
Yeah, I'm sorry for the confusion here, the sorting function prioritizes lesser numbers, so in that way `0` is the highest priority. I should have written in the documentation of the function `getOnboardingPriority` that `999` is the min priority.

Now I'm thinking if we don't want to have an exception thrown here if the onboarding is not found. In the end, if there will be only one onboarding in the array, and it won't exist, it will lead to errors further in the runtime anyway (not existing steps, etc.)

Reply:
Ok that makes more sense now that you explain it, thanks